lib/superagent-mock.js

```diff
@@ -142,7 +142,14 @@ function mock (superagent, config, logger) {
       }
     }
 
-    fn(error, response);
+    if (typeof context.delay === 'number') {
+      setTimeout(function () {
+        fn(error, response);
```

Comment:
Actually response is only used inside the end function, so if we move it inside it will solve it. Do you agree?